File: Lv.2/12909.py
def solution(s):
    stack = 0
    
    if (s[0] == ')' or s[-1] == '('):
        return False
    
    for i in range(len(s)):
        if (s[i] == '('):
            stack += 1
        else:
            stack -= 1
            if (stack < 0):
                return False
            
    if (stack == 0):
        return True

    return False

File: Lv.2/12909.js
function solution(s){
    let stack = [];

    if (s[0] == ')' || s[-1] == '(') {
        return false;
    }
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] == '(') {
            stack.push(s[i]);
        } else {
            stack.pop();
        }
    }
    if (stack.length == 0)
        return true

    return false;
}
